verification.py

```diff
@@ -4623,8 +4629,25 @@ def run(self, output_dir, names=None, group=None):
             tests = [test for test in self.tests if test.group == group]
         else:
             tests = self.tests
-        for test in tests:
-            test.run(output_dir)
+
+        with concurrent.futures.ThreadPoolExecutor(max_workers=threads) as executor:
```

Comment:
Let's make this a ProcessPoolExecutor I think - we'll hit problems with matplotlib otherwise I think.